scenario/context.py

```diff
@@ -38,35 +39,47 @@
 
 
 @dataclasses.dataclass(frozen=True)
-class ActionOutput(_max_posargs(0)):
-    """Wraps the results of running an action event with ``run_action``."""
+class Task(_max_posargs(0)):
+    """Wraps the results of running an action event on a unit with ``run``."""
```

Comment:
Also I'd add it clear in the docstring that this class isn't meant to be used directly. (can we make it private?)

Reply:
Ha, this ends up being the `_Manager` discussion all over again :laughing:. I'm talking to @benhoyt later on today and will have a chat about privacy and docs etc and see where we land up and then can take that to you also.

Reply:
Ok, so Ben convinced me that these all should be/stay public. Reasoning:

* Documentation - I'm not super swayed by this, since it's simple enough to just include the _Private classes in auto generated docs (plus I'm not particularly convinced that objects like this will get looked up). But it is tidier.
* Use in type annotations - I'm not a big fan of type annotations in Python in general, but we offer them so should make that clean. It does seem feasible that someone has a test helper or fixture that takes/returns a `Task` or a `Manager` and so on, and that shouldn't require using a _Private name.
* isinstance checks: the Python purist in me says that these should be rare in favour of duck typing, but the practical voice in my head says that `isinstance(r, RelationBase)` is a reasonable thing to do

My reasoning against is:

* The smaller the API surface the better, and I shouldn't need to know about Manager, Task, RelationBase, and so on - the docs can be smaller, the chunk of my brain that holds "Scenario" can be smaller - and in practice my IDE is going to suggest everything I need to know about these objects.
* I want to keep the ability to change the implementation, including the name, without doing a breaking release. This was my main motivator, and on reflection I should just have more confidence that the choices are fine and that although we're aiming to not need a 8.x any time soon, it's not the end of the world if there is one.

So `Task` remains `Task` and `_Manager` becomes `Manager`. But also to make this consistent `_Port` becomes `Port`, and `_RelationBase` goes back to the 6.x `RelationBase`.

If you're not sufficiently convinced, let's find a time for the three of us to get together and talk it out :).

Reply:
ok ok: can of worms, got it.
I'm convinced as well, let's go for it :)